HIP/hip-0000-mirror-node-account-id.md

```diff
@@ -0,0 +1,68 @@
+- hip: <HIP number (this is determined by the HIP editor)>
+- title: Message Account Id in Mirror Node API Response
+- author: scalemaildev
+- type: Standards Track
+- category: Mirror
+- status: Draft
+- created: 2021-10-18
+- discussions-to: <a URL pointing to the official discussion thread>
+- updated: <comma separated list of dates>
+- requires: <HIP number(s)>
+- replaces: <HIP number(s)>
+- superseded-by: <HIP number(s)>
+
+## Abstract
+
+Add a parameter to the Hedera mirror node API that returns the Account ID of each message's sender in the metadata. This is specific to the message(s) GET request.
+
+## Motivation
+
+In the development of the [Hashgraph Chess Application](https://github.com/scalemaildev/hashgraph_chess) I ran into a problem: the game needed to filter out chess moves and chat messages that aren't from one of the game's two players. But I also didn't want to make either player the admin of their game's topic.
+
+Since the Account ID of a message's sender isn't included in the message metadata, I had to put their ID into the message content. This leaves the application open to spoof attacks, as anyone can submit a message with a false Account ID field. While there are some workaround methods to address this, the simplest method for resolving this problem would be to add a parameter to the mirror node's API that'll return each message's sender's Account ID along with the rest of its metadata during a GET request.
+
+## Rationale
+
+Many use cases for the HCS would want to know who is sending a message to a public topic. Workaround methods, such as querying the Transaction ID for a message and pulling the sender's Account ID from its timestamp, would involve extra API calls and put unnecessary strain on the mirror node and add complexity to the app. Having a trustworthy source of information for who sent a message would allow developers to handle logic involving the sender's ID in their application.
+
+## User stories
+
+As a developer, I want to know what account sent an HCS message from the message's metadata.
+  
+## Specification
+
+The technical specification should describe the syntax and semantics of any new features. The specification should be detailed enough to allow competing, interoperable implementations for at least the current Hedera ecosystem.
```

Comment:
Please fill out these sections to your best knowledge or leave them empty. Please note that the HIP can't be considered complete without Specification.